File: matlab_tools/read_output.m
function [  diagnostic_diffusion_coefficient_ratio, ...
            t, ...
            msd, ...
            msd_x, ...
            msd_y, ...
            msd_z, ...
            execution_time] = read_output(file_path)

file_string = fileread(file_path);

diagnostic_diffusion_coefficient_ratio = read_xml_key(file_string, 'diagnostic_diffusion_coefficient_ratio', 'scalar');
t = read_xml_key(file_string, 'time', 'array');
msd = read_xml_key(file_string, 'mean_square_displacement', 'array');
msd_x = read_xml_key(file_string, 'mean_square_displacement_x', 'array');
msd_y = read_xml_key(file_string, 'mean_square_displacement_y', 'array');
msd_z = read_xml_key(file_string, 'mean_square_displacement_z', 'array');
execution_time = read_xml_key(file_string, 'execution_time', 'scalar');

end

File: src/file_io/read_output.jl
function read_output(file_path::String)
	file_stream::IOStream = open(file_path, "r")
	file_string::String = readstring(file_stream)
	close(file_stream)

	D_empirical::Float64 = read_key(file_string, "D_empirical", Float64)
	t::Array{Float64, 1} = read_key(file_string, "t", Array{Float64, 1})
	msd::Array{Float64, 1} = read_key(file_string, "msd", Array{Float64, 1})
	msd_x::Array{Float64, 1} = read_key(file_string, "msd_x", Array{Float64, 1})
	msd_y::Array{Float64, 1} = read_key(file_string, "msd_y", Array{Float64, 1})
	msd_z::Array{Float64, 1} = read_key(file_string, "msd_z", Array{Float64, 1})
	t_exec::Float64 = read_key(file_string, "t_exec", Float64)

	return (
		D_empirical,
		t,
		msd,
		msd_x,
		msd_y,
		msd_z,
		t_exec)
end
